README.md

```diff
@@ -75,68 +80,113 @@ Use SnowCLI to build a function or stored procedure, or create a streamlit if yo
 
-For example, to add a Snowflake account `myaccount` for a user profile `johndoe` and a password of `hunter2`, add the following:
+   ```
+   $ vi config.toml
+   ```
 
-```ini
-[connections.connection_name]
-accountname = myaccount
-username = jondoe
-password = hunter2
-```
+1. Add a new Snowflake connection definition. You must prefix the configuration with **connections**.
 
-#### Add credentials manually using Windows
+   For example, to add a Snowflake connection called **myconnection** with the credentials: account **myaccount**,
+   user profile **johndoe**, and password **hunter2**,
+   add the following lines to the configuration file:
 
-1. Create a `.snowsql` folder with a `config` file at the following path: `%USERPROFILE%\.snowsql\config`
-2. Add a new configuration for your Snowflake connection with SnowCLI. You must prefix the configuration with `connections.`.
+   ```
+   [connections]
+   [connections.myconnection]
+   accountname = myaccount
+   username = jondoe
+   password = hunter2
+   ```
 
-### Using environment variables
+1. If desired, you can add more connections, as shown:
 
-Environment variables can only be replaced in connection parameters. Parameter values in config file have to start with `$SNOWCLI_` and will be searched as a name in environment variables, but without `$` at the beginning.
+   ```
+   [connections]
+   [connections.myconnection]
+   accountname = myaccount
+   username = jondoe
+   password = hunter2
 
-For example connection config:
+   [connections.myconnection-test]
+   accountname = myaccount
+   username = jondoe-test
+   password = hunter2
+   ```
 
-```ini
-[connections.test]
-account = $SNOWCLI_ACCOUNT
-user = $SNOWCLI_USER
-password = $SNOWCLI_PASSWORD
-```
+1. Save changes to the file.
 
-will be replaced with environment variables:
 
-```ini
-SNOWCLI_ACCOUNT = "account"
-SNOWCLI_USER = "user"
-SNOWCLI_PASSWORD = "password"
-```
+#### Use environment variables for Snowflake credentials
+
+If you prefer, you can specify Snowflake credentials in system environment variables, instead of specifying them
+in configuration files. You can use environment variables only to replace connection parameters. Environment variables for
+the configuration must use the following formats:
+
+- SNOWFLAKE_\<section-name\>_\<key\>=\<value\>
+- SNOWFLAKE_\<section-name\>_\<option-name\>__\<key\>=\<value\>
+
+where:
+
+- \<section-name\> is the name of the section in the configuration file.
+- \<option-name\> is the name of the option in the configuration file.
+- \<key\> is the name of the key
+
+For example: SNOWFLAKE_CONNECTIONS_MYCONNECTION_ACCOUNT="my-account"
+
+
+To add credentials as environment variables:
+
+1. Define the following environment variables, as appropriate for your operating system:
+
+   ```
+   [connections]
+   [connections.myconnection]
+   accountname = SNOWFLAKE_CONNECTIONS_MYCONNECTION_ACCOUNT
```

Comment:
Currently environment variables are not used in the config file at all.

You can set environment variables (in your system/shell, outside of the config file) and they will override values set in configuration. Or environment variables can be combined with the values from the config file.

Examples:

1)) Combination of environment variables and the config file:
Config file:
```
[connections]
[connections.myconnection]
account = myaccount
user = jondoe
```

Environment variable set in system/shell:
```
SNOWFLAKE_CONNECTIONS_MYCONNECTION_PASSWORD=pass1234
```

2)) Overriding values from the config file by environment variables:
```
[connections]
[connections.myconnection]
account = myaccount
user = jondoe
password=xyz000
```

Environment variable set in system/shell:
```
SNOWFLAKE_CONNECTIONS_MYCONNECTION_PASSWORD=pass1234
```

---

Both examples have the same result - password "pass1234".

Reply:
Updated